mdx-layout/src/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import { MDXProvider } from "@mdx-js/react";
+import * as path from "path";
+import { Playground } from "../../playground/dist/index.js";
+import { Props } from '../../props/dist/index.js';
+import { Showcases } from '../../showcases/dist/index.js';
+import { Space } from '../../space/dist/index.js';
+import "@divriots/markdown-github/dist/index.css";
+
+const navItemStyle = {
+  display: "flex",
+  padding: ".5rem 2rem .5rem 0",
+  margin: 0,
+  textTransform: "capitalize",
+} as any;
+
+export const NavHeading = ({ children }) => (
+  <h4 style={navItemStyle}>{children}</h4>
+);
+
+const NavLink = ({ pkg, ...props }) => {
+  const { pages, packages = {} } = props;
+  const page = pages.find(
+    (p) =>
+      p.url === pkg ||
+      p.url.startsWith(`${path.join(packages.dir || "", pkg, "doc")}`)
+  );
+  return page ? (
+    <a
+      style={{ ...navItemStyle, color: "currentcolor", userSelect: "none" }}
+      href={path.join(
+        Array(page.url.split("/").length - 1)
+          .fill("..")
+          .join("/"),
+        page.url
+      )}
```

Comment:
maybe extract a named function for this logic, to explain what it does.

Reply:
done